### ISSUE TITLE
[iceprod2] TestDB makeover (Trac #1432)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1424]
Make testing easier by mocking the DB with TestDB:
- sqlite in-memory
- interface:
  - setup(tables={}) - insert values into tables
  - get(tables=[]) - dump tables
  - get_trace() - every sql call after setup

Migrated from https://code.icecube.wisc.edu/ticket/1432
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:28", 
    "description": "Make testing easier by mocking the DB with TestDB:\n- sqlite in-memory\n- interface:\n  - setup(tables={}) - insert values into tables\n  - get(tables=[]) - dump tables\n  - get_trace() - every sql call after setup", 
    "reporter": "david.schultz", 
    "cc": "ddelventhal, juancarlos", 
    "resolution": "fixed", 
    "_ts": "1550067088921308", 
    "component": "iceprod", 
    "summary": "[iceprod2] TestDB makeover", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2015-11-11T20:49:54", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20151112T18:19:48: 
* david.schultz changed resolution from "" to "fixed"
* david.schultz commented:

> In changeset 139474:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="139474"
> new DB mocking. fixes #1424. see dbmethods_cron_test for a working example
> ```


* david.schultz changed status from "new" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:28: 
* olivas changed milestone from "IceProd 2.0" to ""
* olivas commented:

> Milestone IceProd 2.0 deleted
